### ISSUE TITLE
IBController does not work with TWS 974

### ISSUE THREAD

**Issue by rlktradewright** [ib-controller/ib-controller#207]
It has been brought to my attention (see https://github.com/IbcAlpha/IBC/issues/27) that IBC (and of course IBController) is not currently working with TWS 974, which has recently become IBKR’s ‘latest’ offline TWS.

A brief investigation last night confirmed this.

The problem is due to IBKR having completely re-implemented the login dialog, in such a way that IBC/IBController no longer recognises it, and is therefore unable to inject the username and password into the appropriate fields or click the login button.

It gets worse: because of the way IBKR have done this re-implementation, IBC/IBController’s current technology is not able to explore the detailed structure of the dialog to give it any possibility of finding these things.

In other words, IBC/IBController CANNOT CURRENTLY WORK WITH TWS 974, at least as far as auto-login is concerned.

Therefore I ask you not to bother trying to use IBC/IBController with TWS 974 until further notice. In particular, please do not post here or on the Github repositories about this, as every such post consumes time that I need to try and sort this mess.

I am hopeful that I can get round this problem, but until I investigate further I won’t know. 

The worst case scenario would be that there is no way around this, in which case IBC/IBController will have much less value, and possibly even become useless to most users. I will have more to say about this when I’ve had time to investigate more fully.

Please note that if I manage to fix this problem, I will not be submitting the changes to IBController, only to IBC - see https://github.com/IbcAlpha/IBC.

Thanks for your cooperation.

Richard

**Comment by rlktradewright** [comment]
Further to my earlier message, I’m glad to be able to announce that I have managed to get IBC working with TWS (and Gateway) 974.

It turned out to be more straightforward than I anticipated.

I’ll deploy the new version on my systems over the weekend, and let it run for a few days before releasing it, because there may be other things that have been changed that I haven’t encountered yet.

With any luck I should be able to release a new version towards the end of next week. If anyone is really desperate to use TWS 974 with IBC, I could release a beta version earlier, but I’d rather just do the job properly once.

Note that I won't be applying these changes to IBController, so if you want to work with TWS 974 you'll have to switch to IBC.
